instrumentation/src/main/java/io/opentelemetry/android/instrumentation/crash/CrashReporter.java

```diff
@@ -38,16 +45,47 @@ public void installOn(InstrumentedApplication instrumentedApplication) {
                 Thread.getDefaultUncaughtExceptionHandler();
         Thread.setDefaultUncaughtExceptionHandler(
                 new CrashReportingExceptionHandler(
-                        buildInstrumenter(instrumentedApplication.getOpenTelemetrySdk()),
-                        instrumentedApplication.getOpenTelemetrySdk().getSdkTracerProvider(),
+                        buildInstrumenter(
+                                instrumentedApplication
+                                        .getOpenTelemetrySdk()
+                                        .getSdkLoggerProvider()),
+                        instrumentedApplication.getOpenTelemetrySdk().getSdkLoggerProvider(),
                         existingHandler));
     }
 
-    private Instrumenter<CrashDetails, Void> buildInstrumenter(OpenTelemetry openTelemetry) {
-        return Instrumenter.<CrashDetails, Void>builder(
-                        openTelemetry, "io.opentelemetry.crash", CrashDetails::spanName)
-                .addAttributesExtractor(new CrashDetailsAttributesExtractor())
-                .addAttributesExtractors(additionalExtractors)
-                .buildInstrumenter();
+    private void emitCrashEvent(Logger crashReporter, CrashDetails crashDetails) {
+        Throwable throwable = crashDetails.getCause();
+        Thread thread = crashDetails.getThread();
+        AttributesBuilder attributesBuilder =
+                Attributes.builder()
+                        .put(SemanticAttributes.EXCEPTION_ESCAPED, true)
+                        .put(SemanticAttributes.THREAD_ID, thread.getId())
+                        .put(SemanticAttributes.THREAD_NAME, thread.getName())
+                        .put(SemanticAttributes.EXCEPTION_MESSAGE, throwable.getMessage())
+                        .put(SemanticAttributes.EXCEPTION_STACKTRACE, stackTraceToString(throwable))
+                        .put(SemanticAttributes.EXCEPTION_TYPE, throwable.getClass().getName());
+
+        for (AttributesExtractor<CrashDetails, Void> extractor : additionalExtractors) {
+            extractor.onStart(attributesBuilder, Context.current(), crashDetails);
+        }
+
+        crashReporter.logRecordBuilder().setAllAttributes(attributesBuilder.build()).emit();
+    }
+
+    private String stackTraceToString(Throwable throwable) {
+        StringWriter sw = new StringWriter(256);
+        PrintWriter pw = new PrintWriter(sw);
+
+        throwable.printStackTrace(pw);
+        pw.flush();
+
+        return sw.toString();
+    }
+
+    private Consumer<CrashDetails> buildInstrumenter(LoggerProvider loggerProvider) {
+        return crashDetails ->
+                emitCrashEvent(
+                        loggerProvider.loggerBuilder("io.opentelemetry.crash").build(),
```

Comment:
While I do understand that "crashes" aren't something that happens all the time, it's possible that the consumer here gets called multiple times? It reads a little oddly to me that each time we want to emit a crash event we create a new logger. 

I think instead you should create the logger above the return and then reference it so that it will both front-load the object creation, which should make the consumer more lightweight, and also allows for reuse without having to create an entirely new logger each time.

Reply:
Thanks, that makes sense. I've updated it.